File: backend/app/schemas/analytics.py
from typing import List, Dict, Any
from pydantic import BaseModel, Field

class RiskDistributionItem(BaseModel):
    risk_band: str = Field(..., description="Risk band: Low, Moderate, High, Very High")
    count: int = Field(..., description="Number of cases in this band")
    percentage: float = Field(..., description="Percentage of total cases in this band")

class AnalyticsSummaryResponse(BaseModel):
    total_cases: int
    total_predictions: int
    high_risk_cases_count: int
    high_risk_cases_percentage: float
    average_risk_score: float
    average_predicted_duration_days: float

class CourtRiskItem(BaseModel):
    court_identifier: str
    state_code: str
    dist_code: str
    court_no: str
    case_count: int
    average_risk_score: float
    high_risk_percentage: float
    average_duration_days: float

class CaseTypeRiskItem(BaseModel):
    type_name: str
    case_count: int
    average_risk_score: float
    high_risk_percentage: float

File: frontend/src/types/analytics.ts
import { RiskBand } from './prediction';

export interface AnalyticsSummary {
  total_cases: number;
  total_predictions: number;
  high_risk_cases_count: number;
  high_risk_cases_percentage: number;
  average_risk_score: number;
  average_predicted_duration_days: number;
}

export interface RiskDistributionItem {
  risk_band: RiskBand;
  count: number;
  percentage: number;
}

export interface CourtAnalyticsItem {
  court_identifier?: string;
  court_str?: string;
  state_code?: string;
  dist_code?: string;
  court_no?: string;
  case_count: number;
  high_risk_count?: number;
  high_risk_percentage: number;
  average_risk_score: number;
  average_duration_days?: number;
}

export interface CaseTypeAnalyticsItem {
  type_name: string;
  case_count: number;
  high_risk_count?: number;
  high_risk_percentage: number;
  average_risk_score: number;
}
